athena-gcs/src/main/java/com/amazonaws/athena/connectors/gcs/GcsMetadataHandler.java

```diff
@@ -172,7 +171,6 @@ public GetTableResponse doGetTable(BlockAllocator blockAllocator, GetTableReques
             //fetch schema from dataset api
             Schema schema = datasource.buildTableSchema(table, allocator);
             Map<String, String> columnNameMapping = getColumnNameMapping(table);
-            List<Column> partitionKeys = table.getPartitionKeys() == null ? List.of() : table.getPartitionKeys();
             Set<String> partitionCols = table.getPartitionKeys().stream()
```

Comment:
According to line 175 it is possible that `table.getPartitionKeys()` may return the value null, therefore we should handle that case in the following line where we're setting variable `partitionCols`.

Reply:
Great catch. This was a left over bug where `partitionKeys` should have been used instead of being deleted.